app/src/main/java/com/example/poste/activities/DashboardActivity.java

```diff
@@ -98,37 +101,11 @@ private void createUpdateCallback() {
              */
             @Override
             public void onSuccess() {
-                userFolders = currentUser.getFolders(); // update our local copy of the folders
-                // Fill folder view (Recycler View)
-                // Note: This is not the best way to do this, because it means every time
-                // DashboardActivity restarts, a new Adapter is created, but it works for now. The
-                // better way would be to use DiffUtil to calculate the difference between the old
-                // and new lists, and then update the RecyclerView accordingly using
-                // AdapterListUpdateCallback. This method is an attempt to keep the code simple
-                // For more information, see:
-                // https://developer.android.com/reference/androidx/recyclerview/widget/DiffUtil
-                // and
-                // https://developer.android.com/reference/androidx/recyclerview/widget/ListUpdateCallback
-                folderAdapter = new FolderAdapter(
```

Comment:
Moving this to vars prepping, and simply updating the adapter and notifying of changes